### ISSUE TITLE
grouping shocks in decompositions

### ISSUE THREAD

**Issue by rattoma** [DynareTeam/dynare#1280]
would it be possible to allow a more flexible naming of the groups, allowing a sort of verbatim reading of what is before `=` .
the following syntax does not work
```
shock_groups;
...
Price Mark-up EA = emup;
Wage Mark-up EA = emuw;
...
end;
```
not sure if this can be modified, by allowing group names to be given inside brackets, as follows:
```
shock_groups;
...
(Price Mark-up EA) = emup;
(Wage Mark-up EA) = emuw;
...
end;
```
?

**Comment by houtanb** [comment]
@rattoma how do you propose to have this written in the `.m` file? Currently, the group names are fields in a structure, which cannot accept spaces or operators (in your example `-`). See: http://www.mathworks.com/help/matlab/ref/struct.html#inputarg_field for valid field names in a structure

**Comment by MichelJuillard** [comment]
I think that we should have both a name for the group for Matlab manipulations (maybe de user doesn't need to see it) and a label for display

**Comment by houtanb** [comment]
@MichelJuillard ok. So a proposed output for the above could be:
```
M_.shock_groups.default.group1 = { 'emup'};
M_.shock_groups.default.group1.name = { 'Price Mark-up EA'};
M_.shock_groups.default.group2 = { 'emuw'};
M_.shock_groups.default.group2.name = { 'Wage Mark-up EA'};
```
@rattoma @MichelJuillard, does this sound good?

**Comment by MichelJuillard** [comment]
@houtanb I don't think that you can have
```
group1 = {}
```
and 
```
gorup1. name = xxxx
```
I would prefer
```
group1.shocks = {'emup'}
group1.label = 'Price Mark-up EA'
```

**Comment by houtanb** [comment]
@MichelJuillard you're right. Wrote it too quickly. Will do

**Comment by rattoma** [comment]
@houtanb @MichelJuillard many thanks. I agree with your solution.

**Comment by houtanb** [comment]
NB: Notation changed to be:
```
shock_groups;
...
'Price Mark-up EA'= emup;
'Wage Mark-up EA' = emuw;
...
end;
```

**Comment by rattoma** [comment]
I am going to pull request at this point a new utility
`plot_shock_decomposition`
that allows to plot at a later stage any kind grouping defined using information stored in 
`oo_.shock_decomposition`.

To this end it would be useful to allow for the option nograph into shock_decomposition that filters out the call to `graph_decomp` if one wants to produce customized plots with the new utility.
would this be possible?

**Comment by JohannesPfeifer** [comment]
@rattoma Do you want this as a separate local option to `shock_decomposition`? Or do you simply want an interface to set `options_.nograph`, which would affect later calls to plotting as well?

**Comment by rattoma** [comment]
this is an issue, in fact. Because there is no interface to revert the global `options_.nograph` in a subsequent call to e.g. stoch_simul. So, maybe better a local option?

**Comment by houtanb** [comment]
@rattoma @JohannesPfeifer I don't see the relation of your conversation here to the issue topic. Can you please create a new issue?

**Comment by rattoma** [comment]
@houtan. Concerning bug fixed in fe7b781, I think this requires this issue to re-opened and closed after merging the fix. thanks
